src/github.com/matrix-org/dendrite/clientapi/httputil/parse.go

```diff
@@ -13,26 +13,49 @@
 package httputil
 
 import (
+	"fmt"
 	"net/http"
 	"strconv"
 	"time"
+
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/util"
 )
 
+// ParseIntParam takes a request and parses the query param to an int
+// returns 0 when not present and -1 when parsing failed as int
+func ParseIntParam(req *http.Request, param string) (int64, *util.JSONResponse) {
```

Comment:
Will this be used somewhere ?

I'd rather not have unnecessary Public functions.

Reply:
removed it now. To reintroduce it it is possible to revert https://github.com/matrix-org/dendrite/pull/576/commits/767ad6e36f9b2a5585006510d2ba18e3ac745078